src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/ValuationInformationPolicyServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+
+    private static final String VALUATION_INFORMATION_POLICY_FIELD_PATH =
+            "valuationInformationPolicyDetails";
+
+    private static final String INVALID_STRING_SIZE_ERROR_MESSAGE =
+            "validation.length.minInvalid.accounting_policies.valuation_information_and_policy";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ValuationInformationPolicy getValuationInformationPolicy(String transactionId,
+            String companyAccountsId) throws ServiceException {
+
+        AccountingPoliciesApi accountingPoliciesApi =
+                accountingPoliciesService.getAccountingPoliciesApi(transactionId, companyAccountsId);
+
+        return accountingPoliciesTransformer.getValuationInformationPolicy(accountingPoliciesApi);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<ValidationError> submitValuationInformationPolicy(String transactionId,
+            String companyAccountsId, ValuationInformationPolicy valuationInformationPolicy)
+            throws ServiceException {
+
+        List<ValidationError> validationErrors = validateValuationInformationPolicy(valuationInformationPolicy);
+        if (!validationErrors.isEmpty()) {
+            return validationErrors;
+        }
+
+        AccountingPoliciesApi accountingPoliciesApi =
+                accountingPoliciesService.getAccountingPoliciesApi(transactionId, companyAccountsId);
+
+        accountingPoliciesTransformer
+                .setValuationInformationPolicy(valuationInformationPolicy, accountingPoliciesApi);
+
+        return accountingPoliciesService
+                .updateAccountingPoliciesApi(transactionId, companyAccountsId, accountingPoliciesApi);
+    }
+
+    /**
+     * Validate a submitted valuation information policy
+     * @param valuationInformationPolicy The policy to validate
+     * @return a list of validation errors, or an empty array list if none are present
+     */
+    private List<ValidationError> validateValuationInformationPolicy(ValuationInformationPolicy valuationInformationPolicy) {
+
+        List<ValidationError> validationErrors = new ArrayList<>();
+
+        if (valuationInformationPolicyNotProvided(valuationInformationPolicy)) {
+            ValidationError validationError = new ValidationError();
+            validationError.setFieldPath(VALUATION_INFORMATION_POLICY_FIELD_PATH);
+            validationError.setMessageKey(INVALID_STRING_SIZE_ERROR_MESSAGE);
+            validationErrors.add(validationError);
+        }
+
+        return validationErrors;
+    }
+
+    /**
+     * Check whether a valuation information has not been provided
+     * @param valuationInformationPolicy The policy to check
+     * @return true if the 'include policy' boolean is set to true, but no details are provided
+     */
+    private boolean valuationInformationPolicyNotProvided(ValuationInformationPolicy valuationInformationPolicy) {
```

Comment:
I would rename this method to something like `isValuationInformationPolicyNotProvided`. What do you think?

Reply:
I found the naming here confusing too. The convention here would be to use a `has` or `is` prefix as Marco has suggested. Perhaps flipping the logic would make the code more readable rather than including 'Not` in the method name, for example:

```
if (!hasValuationInformationPolicy) { ... }
```

Reply:
agree with the naming that Marco recommends, you can then remove the JavaDoc from the private method as the method is self explanatory.

Reply:
Method has been renamed to Marco's recommendation